src/transformers/modeling_utils.py

```diff
@@ -1995,72 +1996,67 @@ def save_pretrained(
+                        logger.info(
+                            "To match the expected format of the PEFT library, all keys of the state dict of adapters will be pre-pended with `base_model.model`."
+                        )
+                        peft_state_dict = {}
+                        for key, value in adapter_state_dict.items():
+                            peft_state_dict[f"base_model.model.{key}"] = value
+                        adapter_state_dict = peft_state_dict.copy()
+                        # Free memory
+                        del peft_state_dict
+
+                    current_peft_config = self.peft_config[adapter_name]
+                    peft_multi_adapter_state_dict[adapter_name] = adapter_state_dict
+
+                    # the default adapter is always saved on the root directory
+                    if adapter_name != "default":
+                        current_peft_config.save_pretrained(os.path.join(save_directory, adapter_name))
+                    else:
+                        current_peft_config.save_pretrained(save_directory)
+
+                if len(peft_multi_adapter_state_dict.keys()) == 1:
+                    current_adapter = list(peft_multi_adapter_state_dict.keys())[0]
+                    state_dict = peft_multi_adapter_state_dict[current_adapter].copy()
+                    peft_multi_adapter_state_dict = None
 
-                current_peft_config = self.peft_config[self.active_adapter()]
-                current_peft_config.save_pretrained(save_directory)
+        _peft_save_multi_adapter = _hf_peft_config_loaded and peft_multi_adapter_state_dict is not None
 
         # Save the model
-        if state_dict is None:
+        if state_dict is None and not _peft_save_multi_adapter:
             state_dict = model_to_save.state_dict()
 
         # Translate state_dict from smp to hf if saving with smp >= 1.10
-        if IS_SAGEMAKER_MP_POST_1_10:
+        if IS_SAGEMAKER_MP_POST_1_10 and not _peft_save_multi_adapter:
             for smp_to_hf, _ in smp.state.module_manager.translate_functions:
                 state_dict = smp_to_hf(state_dict)
 
         # Handle the case where some state_dict keys shouldn't be saved
         if self._keys_to_ignore_on_save is not None:
-            for ignore_key in self._keys_to_ignore_on_save:
-                if ignore_key in state_dict.keys():
-                    del state_dict[ignore_key]
+            if not _peft_save_multi_adapter:
+                for ignore_key in self._keys_to_ignore_on_save:
+                    if ignore_key in state_dict.keys():
+                        del state_dict[ignore_key]
+            else:
+                for adapter_name in peft_multi_adapter_state_dict:
+                    for ignore_key in self._keys_to_ignore_on_save:
+                        if ignore_key in peft_multi_adapter_state_dict[adapter_name].keys():
+                            del peft_multi_adapter_state_dict[adapter_name][ignore_key]
```

Comment:
I asked ChatGPT to make this block more elegant, here is what it came up with:

```python
if not _peft_save_multi_adapter:
    state_dict = {k: v for k, v in state_dict.items() if k not in self._keys_to_ignore_on_save}
else:
    peft_multi_adapter_state_dict = {
        adapter_name: {k: v for k, v in adapter.items() if k not in self._keys_to_ignore_on_save}
        for adapter_name, adapter in peft_multi_adapter_state_dict.items()
    }
```

WDYT? :)